concepts/eventing-code-guidelines.md

```diff
@@ -718,6 +721,119 @@ func TestSomething(t *testing.T) {
 
 </details>
 
+<details>
+  <summary>2-dimensional table-driven test</summary>
+
+There may be situation when you need a table-driven test with more than one dimension. The following shows a best practice example for a 2-dimensional table-driven test.
+
+```go
+func TestTwoDimensions(t *testing.T) {
+	// first dimension
+	testCases := []struct {
+		name               string
+		givenSender        CESender
+		wantHTTPStatusCode int
+	}{
+		{
+			name:        "binary cloud event sender",
+			givenSender: myCEBinarySender,
+		},
+	}
+	// second dimension
+	cloudEvents := []struct {
+		name, givenCEType string
+	}{
+		{
+			name:               "proper cloud event",
+			givenCEType:        "order.created.v1",
+			wantHTTPStatusCode: http.StatusOK,
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			for _, ce := range cloudEvents {
+				ce := ce
+				t.Run(ce.name, func(t *testing.T) {
+					t.Parallel()
+					res := functionUnderTest(tc.givenSender, ce.givenCEType)
+					// check res == ce.wantHTTPStatusCode
+				})
+			}
+		})
+	}
+}
+```
+
+<details>
+  <summary>Reason for variable reassignment (tc := tc)</summary>
+
+The loop iteration variable in Go is a single variable. The closure for the second t.Run is executed inside a goroutine. 
+If there are multiple entries in the `cloudEvents` struct, `ce` references the last entry in `cloudEvents` in every iteration. The problem occurs because the closure referring to `tc` and `ce` is not executed in **sync** with the **for loop** (because of t.parallel).
+To prevent this problem, `ce` and `tc` must be copied (`ce := ce`).
+The linter [scopelint](https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/scopelint.go) warns about the possible problem whenever `ce` or `tc` is used.
+Adding `// nolint:scopelint` to silence scopelint must be used with caution.
```

Comment:
What bad things can happen if you use `// nolint:scopelint` and choose to get no warning? I'd suggest a warning like this:

```suggestion
>**CAUTION:** If you add `// nolint:scopelint` to silence scopelint, you might not notice when `ce` references {a wrong/different??} entry. Use it with caution and make sure to check {...?}.
```